dashboard/app/controllers/aichat_controller.rb

```diff
@@ -0,0 +1,37 @@
+class AichatController < ApplicationController
+  # POST /aichat/chat_completion
+  def chat_completion
+    unless has_required_params?
+      return render(status: :bad_request, json: {})
+    end
+    # Check for PII / Profanity
+    # Copied from ai_tutor_interactions_controller.rb - not sure if filtering is working.
+    locale = params[:locale] || "en"
+    # Check only the newest message from the user for inappropriate content.
+    chatContext = params[:chatContext]
+    newMessageText = chatContext["userMessage"]   
+    filter_result = ShareFiltering.find_failure(newMessageText, locale) if newMessageText
+    # If the content is inappropriate, we skip sending to endpoint and instead hardcode a warning response on the front-end.   
+    return render(status: :ok, json: {status: filter_result.type, flagged_content: filter_result.content}) if filter_result
+    inputs = params[:inputs]
+    temperature = params[:temperature]
+    # TODO: Format input to send to Sagemaker.
```

Comment:
Still waiting clarification on how to format. Using [this ref](https://aws.amazon.com/blogs/machine-learning/best-prompting-practices-for-using-the-llama-2-chat-llm-through-amazon-sagemaker-jumpstart/) as a reference for now.